openiked/patches/42-local_addr.patch

```diff
@@ -0,0 +1,46 @@
+diff -ruN a/iked/iked.c b/iked/iked.c
+--- a/iked/iked.c	2022-01-28 10:16:06.990394016 +0100
++++ b/iked/iked.c	2022-01-31 10:39:14.250484263 +0100
+@@ -34,6 +34,7 @@
+ #include <pwd.h>
+ #include <event.h>
+ #include <fcntl.h>
++#include <ifaddrs.h>
+ 
+ #include "iked.h"
+ #include "ikev2.h"
+@@ -289,6 +290,14 @@
+ parent_configure(struct iked *env)
+ {
+ 	struct sockaddr_storage	 ss;
++#ifdef __phoenix__
++	struct ifaddrs *ifaddr, *ifa;
++
++	if (getifaddrs(&ifaddr) < 0) {
++		proc_kill(&env->sc_ps);
++		exit(1);
++	}
++#endif
+ 
+ 	if (parse_config(env->sc_conffile, env) == -1) {
+ 		proc_kill(&env->sc_ps);
+@@ -315,6 +324,19 @@
+ 	bzero(&ss, sizeof(ss));
+ 	ss.ss_family = AF_INET;
+ 
++#ifdef __phoenix__
++	for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
++		if (strcmp("en1", ifa->ifa_name) == 0) {
```

Comment:
kindof ugly to hardcode `en1` as default interface(?) but I don't know the context enough to provide better alternative

Reply:
Yes, I know it's ugly. I was trying to fix this in a clean way last few hours but couldn't do it. On Andra it seems to work by chance because of non-standard implementation of getsockname().